pkg/pebinaryexecutor/pebinary.go

```diff
@@ -254,6 +277,15 @@ func getArgs(argsMap map[string]string) []string {
 	return args
 }
 
+func argIndex(args []string, s string) (int, error) {
+	for i, arg := range args {
+		if strings.Contains(arg, s) {
+			return i, nil
+		}
+	}
```

Comment:
If you're going to make a function for this, it should be safe to use for any arg - not just ones that you know aren't substrings of other args or possible arg values.
```suggestion
	s = "--" + s
	for i, arg := range args {
		parts := strings.SplitN(arg, "=", 2)
		if strings.HasPrefix(parts[0], s) {
			return i, nil
		}
	}
```

Reply:
This adds functionality beyond what is required at the moment. I am happy to add an issue to track it, I just feel that it adds too much to the scope to address the current need and may cause potential issues if extra logic is added. This is only be used in the context of the Windows agent and the binary executor.

Reply:
I saw you had an earlier version of this that didn't use a function. If you have a function, make it work right and be reusable. If it's not reusable, or is implemented such that it only works for your specific use case, don't make it a function.

Reply:
@brandond function removed.